src/SparkTestBase.sol

```diff
@@ -40,22 +45,133 @@ interface IExecutable {
+            } catch {
+                console.log("skipping spell deployment for network: ", id.toDomainString());
+            }
+        }
+    }
+
+    /// @dev takes care to revert the selected fork to what was chosen before
+    function executeAllPayloadsAndBridges() internal {
+        // only execute mainnet payload
+        executeMainnetPayload();
+        // then use bridges to execute other chains' payloads
+        relayMessageOverBridges();
+    }
+
+    /// @dev bridge contracts themselves are stored on mainnet
+    function relayMessageOverBridges() private onChain(ChainIdUtils.Ethereum()) {
+        // Base
+        OptimismBridgeTesting.relayMessagesToDestination(chainSpellMetadata[ChainIdUtils.Base()].bridge, false);
+        OptimismBridgeTesting.relayMessagesToSource(chainSpellMetadata[ChainIdUtils.Base()].bridge, false);
+        // Gnosis: TODO
+    }
+
+    function executeL2PayloadsFromBridges() private onChain(ChainIdUtils.Ethereum()) {
+        // Base
+        chainSpellMetadata[ChainIdUtils.Base()].domain.selectFork();
+        IExecutor(Base.SPARK_EXECUTOR).execute(IExecutor(Base.SPARK_EXECUTOR).actionsSetCount() - 1);
+        // Gnosis: TODO
+    }
+
+    function executeMainnetPayload() internal onChain(ChainIdUtils.Ethereum()){
+        address payloadAddress = chainSpellMetadata[ChainIdUtils.Ethereum()].payload;
+        IExecutor executor     = chainSpellMetadata[ChainIdUtils.Ethereum()].executor;
+        require(Address.isContract(payloadAddress), "PAYLOAD IS NOT A CONTRACT");
+
+        vm.prank(Ethereum.PAUSE_PROXY);
+        (bool success,) = address(executor).call(abi.encodeWithSignature(
+            'exec(address,bytes)',
+            payloadAddress,
+            abi.encodeWithSignature('execute()')
+        ));
+        require(success, "FAILED TO EXECUTE PAYLOAD");
+    }
+}
+
+/// @dev assertions that make sense to run on every chain where a spark spell
+/// can be executed
+abstract contract CommonSpellAssertions is SpellRunner {
+    function test_ETHEREUM_PayloadBytecodeMatches() internal {
+        testPayloadBytecodeMatches(ChainIdUtils.Ethereum());
+    }
 
-    string internal domain;
-    string internal id;
+    function test_BASE_PayloadBytecodeMatches() internal {
+        testPayloadBytecodeMatches(ChainIdUtils.Base());
+    }
 
-    function deployPayload() internal returns (address) {
-        string memory fullName = string(abi.encodePacked('Spark', domain, '_', id));
-        return deployCode(string(abi.encodePacked(fullName, '.sol:', fullName)));
+    function test_GNOSIS_PayloadBytecodeMatches() internal {
+        testPayloadBytecodeMatches(ChainIdUtils.Gnosis());
     }
 
-    function testPayloadBytecodeMatches() public {
-        require(Address.isContract(payload), "PAYLOAD IS NOT A CONTRACT");
-        address expectedPayload = deployPayload();
-        address actualPayload   = payload;
+    function testPayloadBytecodeMatches(ChainId chainId) internal onChain(chainId) {
```

Comment:
You should not prefix with `test` for a function that is not expected to be ran. It is not running because it is `internal`, but should also not use the test prefix.

Reply:
makes sense, thanks for pointing it out. Applied my best judgement for renaming the functions in 8dfa142